ragmac_xdem/dem_postprocessing.py

```diff
@@ -358,24 +355,31 @@ def postprocessing_single(
     # Filter gross outliers in stable terrain
     inlier_mask = nmad_filter(ddem.data, stable_mask, verbose=False)
     outlier_mask = ~inlier_mask & stable_mask
-    ddem.data.mask[outlier_mask] = True
-    del inlier_mask
 
-    # Calculate coverage on and off ice
-    roi_coverage_orig, nstable_orig, med_orig, nmad_orig = calculate_stats(ddem, roi_mask, stable_mask)
+    # Exclude steep slopes for coreg
+    slope = xdem.terrain.slope(ref_dem)
+    inlier_mask[slope.data > 50] = False
+
+    # Calculate dDEM statistics on pixels used for coreg
+    ddem_inlier = ddem.copy()
+    ddem_inlier.data.mask[outlier_mask] = True
+    roi_coverage_orig, nstable_orig, med_orig, nmad_orig = calculate_stats(ddem_inlier, roi_mask, stable_mask)
+    del ddem_inlier
 
     # Coregister to reference - Note: this will spread NaN
     # Better strategy: calculate shift, update transform, resample
     if isinstance(coreg_method, xdem.coreg.Coreg):
-        # coreg_method = xdem.coreg.NuthKaab() + xdem.coreg.BiasCorr(bias_func=np.nanmedian)
-        coreg_method.fit(ref_dem, dem, stable_mask, verbose=verbose)
+        coreg_method.fit(ref_dem, dem, inlier_mask, verbose=verbose)
```

Comment:
the mask was only set to `stable_mask` rather than `inlier_mask` which further include large outliers.